src/main/java/frc/trigon/robot/subsystems/swerve/PoseEstimator.java

```diff
@@ -0,0 +1,133 @@
+package frc.trigon.robot.subsystems.swerve;
+
+import edu.wpi.first.math.estimator.SwerveDrivePoseEstimator;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.trigon.robot.RobotContainer;
+import frc.trigon.robot.robotposesources.PoseSourceConstants;
+import frc.trigon.robot.robotposesources.RobotPoseSource;
+import io.github.oblarg.oblog.Loggable;
+import io.github.oblarg.oblog.annotations.Log;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+
+public class PoseEstimator extends SubsystemBase implements Loggable {
+    private final static PoseEstimator INSTANCE = new PoseEstimator();
+
+    private final Swerve swerve = RobotContainer.SWERVE;
+    private final SwerveDrivePoseEstimator swerveDrivePoseEstimator;
+    @Log
```

Comment:
If I'm not mistaken this annotation is unnecessary as this class implements `Loggable`

Reply:
The implementation is only there so you could use the `log` and `config` annotations in the class
It doesn't log all fields